src/solid/Virtualizer.tsx

```diff
@@ -133,29 +138,20 @@ export interface VirtualizerProps<T> {
  */
 export const Virtualizer = <T,>(props: VirtualizerProps<T>): JSX.Element => {
   let containerRef: HTMLDivElement | undefined;
-
-  const {
-    ref: _ref,
-    data: _data,
-    children: _children,
-    overscan: _overscan,
-    itemSize,
-    shift: _shift,
-    horizontal = false,
-    onScroll: _onScroll,
-    onScrollEnd: _onScrollEnd,
-    onRangeChange: _onRangeChange,
-  } = props;
+  props = mergeProps<[Partial<VirtualizerProps<T>>, VirtualizerProps<T>]>(
```

Comment:
I'm not sure why this was being done here because I didn't see those renamed props being used? 🤔 

Also, I think destructuring isn't allowed in solidjs if we want to maintain reactivity. To add defaults, they recommend using `mergeProps` and to destructure we can use `splitProps`

Reply:
There were 2 reasons:

- horizontal and itemSize prop is not reactive because store/scroller/resizer are initilized only once
- To support `<div {...rest}/>` like case in VList https://github.com/solidjs/solid/issues/75 .  It was also accidentally copied to Virtualizer so it's not necessary!

So remaining only horizontal and itemSize may be better.

```suggestion
  const {
    itemSize,
    horizontal = false,
  } = props;
  props = mergeProps<[Partial<VirtualizerProps<T>>, VirtualizerProps<T>]>(
```